iTunesSearch/iTunesSearch/Controller/SearchResultController.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  SearchResultController.swift
+//  iTunesSearch
+//
+//  Created by Kobe McKee on 5/14/19.
+//  Copyright © 2019 Kobe McKee. All rights reserved.
+//
+
+import Foundation
+
+class SearchResultController {
+    
+    let baseURL = URL(string: "https://itunes.apple.com/search")!
+    
+    enum HTTPMethod: String {
+        case get = "GET"
+        case put = "PUT"
+        case post = "POST"
+        case delete = "DELETE"
+    }
+    
+    var results: [SearchResult] = []
+    
+    
+    func performSearch(searchTerm: String, resultType: ResultType, completion: @escaping (Error?) -> Void) {
+        
+        var urlComponents = URLComponents(url: baseURL, resolvingAgainstBaseURL: true)
+        let searchQueryItem = URLQueryItem(name: "term", value: searchTerm)
+        //https://itunes.apple.com/search?term=(searchTerm)
+        
+        let resultTypeQueryItem = URLQueryItem(name: "entity", value: resultType.rawValue)
+        //https://itunes.apple.com/search?term=(searchTerm)&entity=(resultType)
+        
+        urlComponents?.queryItems = [searchQueryItem, resultTypeQueryItem]
+        
+        
+        guard let formattedURL = urlComponents?.url else {
+            completion(nil)
```

Comment:
In the completion there if the guard fails you don't want to say nil, nil means nothing. Instead you can create an error to let wherever this function is called know of the error. 

 `completion(NSError())`

NSError will create the error for you.